src/openapi-generator.ts

```diff
@@ -103,7 +113,13 @@ export class OpenAPIGenerator {
     };
   }
 
-  generateComponents(): Pick<OpenAPIObject, 'components'> {
+  generateComponents(
+    config?: Pick<OpenAPIObjectConfig, 'openapi'>
```

Comment:
I would personally remove this - same reasoning as above. If we do keep it - I feel like we should rather keep this as a simple schema - `{ openapi?: OpenApiVersion }` - no need to "link" it to the OpenAPIObjectConfig